pkg/cmd/passholetemplatefuncs.go

```diff
@@ -0,0 +1,127 @@
+package cmd
+
+import (
+	"bytes"
+	"errors"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/mitchellh/go-homedir"
+
+	"github.com/twpayne/chezmoi/v2/pkg/chezmoi"
+	"github.com/twpayne/chezmoi/v2/pkg/chezmoilog"
+)
+
+type passholeConfig struct {
+	Command      string
+	Args         []string
+	Config       chezmoi.AbsPath
+	Database     chezmoi.AbsPath
+	Keyfile      chezmoi.AbsPath
+	NoPassword   bool
+	NoCache      bool
+	CacheTimeout int
+	cache        map[string]string
+	password     string
+}
+
+func raise(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+func PathExists(path string) bool {
+	_, err := os.Stat(path)
+	return err == nil
+}
+
+func (c *Config) passholeAssembleArgs(args []string) []string {
+	var result []string
+	if c.Passhole.Config.Empty() && c.Passhole.Database.Empty() {
+		panic(errors.New("passhole.database or passhole.config not set"))
+	}
+	if !c.Passhole.Config.Empty() {
+		result = append([]string{"--config", c.Passhole.Config.String()}, args...)
+		return result
+	}
+	switch {
+	case !c.Passhole.Database.Empty():
+		result = append([]string{"--database", c.Passhole.Database.String()}, args...)
+	case !c.Passhole.Keyfile.Empty():
+		result = append([]string{
+			"--database",
+			c.Passhole.Database.String(), "--keyfile", c.Passhole.Keyfile.String(),
+		}, args...)
+	default:
+		panic(errors.New("passhole.database not set"))
+	}
+	if c.Passhole.NoCache {
+		result = append([]string{"--no-cache"}, result...)
+	}
+	if c.Passhole.NoPassword {
+		result = append([]string{"--no-password"}, result...)
+	}
+	return result
+}
+
+func (c *Config) passholeTemplateFunction(record string) map[string]string {
+	home, err := homedir.Dir()
+	raise(err)
+	defaultConfigPath := home + "/.config/passhole.ini"
```

Comment:
This assumes UNIX-style paths. chezmoi has `chezmoi.AbsPath.Join` for joining paths in a OS-independent way.